pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -338,9 +338,26 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
+			firstlinkdstPath, err := os.Readlink(logSymlink)
 			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+				glog.Errorf("Failed to resolve the log symlink %q: %v", logSymlink, err)
+				return nil
+			}
+			secondlinkdstPath, err := os.Readlink(firstlinkdstPath)
+			if err != nil {
+				glog.Errorf("Failed to resolve the log firstlinkdstPath %q: %v", firstlinkdstPath, err)
+				return nil
+			}
+			dir := filepath.Dir(secondlinkdstPath)
+			if _, err := osInterface.Stat(dir); os.IsNotExist(err) {
+				err1 := osInterface.Remove(logSymlink)
+				if err1 != nil {
+					glog.Errorf("Failed to remove container log dead firstsymlink %q: %v", logSymlink, err1)
+				}
+				err2 := osInterface.Remove(firstlinkdstPath)
```

Comment:
I think you don't need to handle the firstlinkdstPath which under path /var/log/pods/, the firstlinkdstPath is related to the existence of pod, and if i am not missing something, it has been handled by other mechanism of kubelet.

Reply:
@xialonglee Can you specify the mechanism. I've made practice ,when I use "docker rm container",this firstlinkdstPath will be residual.

Reply:
`firstlinkdstPath` could be removed through check `func (ds *dockerService) removeContainerLogSymlink` when the pod has been removed.

But, you are right, i retest again, those mechanism didn't cover all situation especially when we use `docker` command to operate these containers in the pod directly.

`firstlinkdstPath` should be handled.